fast64_internal/mk64/__init__.py

```diff
@@ -0,0 +1,138 @@
+import math
+import bpy
+from bpy_types import PropertyGroup
+import mathutils
+from ..f3d.f3d_gbi import get_F3D_GBI
+from ..f3d.f3d_material import createF3DMat
+from ..f3d.f3d_parser import F3DContext, getImportData, importMeshC
+from ..panels import MK64_Panel
+from ..utility import prop_split
+from bpy.utils import register_class, unregister_class
+from ..utility import raisePluginError
+from .f3d_course_parser import processCommands, parseCourseVtx
+
+class MK64_Properties(PropertyGroup):
+    """Global MK64 Scene Properties found under scene.fast64.mk64"""
+
+    version: bpy.props.IntProperty(name="MK64_Properties Version", default=0)
```

Comment:
FTR this way of versioning is (always has been) broken, cf https://github.com/Fast-64/fast64/issues/179

The way we do it now is simply removing the property definitions and checking the data blocks for previous data, for example:
```py
# before
SomeDataType.my_number = IntProperty()

# after
SomeDataType.my_number_renamed = IntProperty()
def upgrade_props(data: SomeDataType): # runs unconditionally on opening a blend
    if "my_number" in data:
        my_number_old_val = data["my_number"]
        data.my_number_renamed = my_number_old_val
        del data["my_number"]
```

@Yanis42 @kurethedead could you confirm this is correct?

Reply:
it's not upgrade_changed_props ? I saw that for sm64 but don't know how it's work exaclty and after I see OOT_Properties who do like I do so I just need to switch to upgrade_changed_props ?

Reply:
I put it for the futur when version will mater

Reply:
yeah but what I'm trying to express here is that using a version property does not work for upgrading props

Reply:
so how I do it's fine ? (after the adding upgrade_changed_props)

Reply:
Still using a `version` property so no

Reply:
so how it should be ? currently it's like this:
```python
class MK64_Properties(PropertyGroup):
    """Global MK64 Scene Properties found under scene.fast64.mk64"""

    version: bpy.props.IntProperty(name="MK64_Properties Version", default=0)
    cur_version = 0

    # Import Course DL
    EnableRenderModeDefault: bpy.props.BoolProperty(name="Enable Render Mode by Default", default=True)

    @staticmethod
    def upgrade_changed_props():
        if bpy.context.scene.fast64.mk64.version != MK64_Properties.cur_version:
            bpy.context.scene.fast64.mk64.version = MK64_Properties.cur_version
```

Reply:
it should be like in my first comment above, which means there should be nothing since currently there's nothing to upgrade

for the 5th time, using a version property does not work!

Reply:
I mainly copy what sm64 do

Reply:
I hardly see what SomeDataType is and by seeing how Oot and sm64 do I don't see the pattern describe in the first comment

Reply:
I am not familiar on how blender work can you give me the link in the doc or a example in code of fast64 who are how you describe ?

Reply:
this PR fixes upgrading of old props `dontTransition` and `roomIndex` in oot data
https://github.com/Fast-64/fast64/pull/287/files

Reply:
of what I see it's just update attribute of a class so like here nothing change it's not needed ? because he are not base on a version just of what attribute he have

Reply:
so have this for now:
```python
class MK64_Properties(PropertyGroup):
    """Global MK64 Scene Properties found under scene.fast64.mk64"""
    # Import Course DL
    EnableRenderModeDefault: bpy.props.BoolProperty(name="Enable Render Mode by Default", default=True)

    @staticmethod
    def upgrade_changed_props(_:MK64_Properties):
        pass
```